internal/graph/check.go

```diff
@@ -91,8 +91,9 @@ const (
 )
 
 type checkOutcome struct {
-	resp *openfgapb.CheckResponse
```

Comment:
Instead of adding an additional return value to the Check signatures, you should add this as additional metadata in the `ResolutionMetadata` structure. The approach taken here won't work when dispatching is introduced later, and the purpose behind the current design is to allow the implementation to be flexible in that regard.

If you add `DatabaseReadCounter` to the `ResolutionMetadata` structure and increment it anytime you do a read, then if we dispatch locally it will work the same as if we dispatch remotely.

Reply:
If I do this, I'll have to  change

```go
type ResolveCheckResponse struct {
	Allowed bool
}
```

to 

```go
type ResolveCheckResponse struct {
	Allowed            bool
	ResolutionMetadata *ResolutionMetadata
}
```

which i'm fine with. But, it will be confusing that the resolution metadata is also in the request object...


```go
type ResolveCheckRequest struct {
	StoreID              string
	AuthorizationModelID string
	TupleKey             *openfgapb.TupleKey
	ContextualTuples     []*openfgapb.TupleKey
	ResolutionMetadata   *ResolutionMetadata
}
```

Reply:
I feel like this change is a good candidate for a pair programming session